referenceframe/worldstate.go

```diff
@@ -58,51 +77,71 @@ func WorldStateToProtobuf(worldState *WorldState) (*commonpb.WorldState, error)
 		return list
 	}
 
-	transforms, err := LinkInFramesToTransformsProtobuf(worldState.Transforms)
+	transforms, err := LinkInFramesToTransformsProtobuf(ws.transforms)
 	if err != nil {
 		return nil, err
 	}
 
 	return &commonpb.WorldState{
-		Obstacles:  convertGeometriesToProto(worldState.Obstacles),
+		Obstacles:  convertGeometriesToProto(ws.obstacles),
 		Transforms: transforms,
 	}, nil
 }
 
+// ObstacleNames returns the set of geometry names that have been registered in the WorldState, represented as a map.
+func (ws *WorldState) ObstacleNames() map[string]bool {
+	if ws == nil {
+		return map[string]bool{}
+	}
+
+	copiedMap := make(map[string]bool)
+	for key, value := range ws.obstacleNames {
+		copiedMap[key] = value
+	}
+	return copiedMap
+}
+
+// Transforms returns the transforms that have been added to the WorldState.
+func (ws *WorldState) Transforms() []*LinkInFrame {
+	if ws == nil {
+		return []*LinkInFrame{}
+	}
+	return ws.transforms
+}
+
 // ObstaclesInWorldFrame takes a frame system and a set of inputs for that frame system and converts all the obstacles
 // in the WorldState such that they are in the frame system's World reference frame.
 func (ws *WorldState) ObstaclesInWorldFrame(fs FrameSystem, inputs map[string][]Input) (*GeometriesInFrame, error) {
-	transformGeometriesToWorldFrame := func(gfs []*GeometriesInFrame) (*GeometriesInFrame, error) {
-		nameCheck := make(map[string]bool)
-		allGeometries := make([]spatial.Geometry, 0, len(gfs))
-
-		unnamedCount := 1
-
-		for _, gf := range gfs {
-			tf, err := fs.Transform(inputs, gf, World)
-			if err != nil {
-				return nil, err
-			}
-			for _, g := range tf.(*GeometriesInFrame).Geometries() {
-				geomName := g.Label()
-				if geomName == "" {
-					geomName = unnamedWorldStateGeometryPrefix + strconv.Itoa(unnamedCount)
-					g.SetLabel(geomName)
-					unnamedCount++
-				}
-
-				if _, present := nameCheck[geomName]; present {
-					return nil, fmt.Errorf("cannot specify multiple geometries with the same name: %s", geomName)
-				}
-				nameCheck[geomName] = true
-				allGeometries = append(allGeometries, g)
-			}
+	if ws == nil {
+		return NewGeometriesInFrame(World, []spatialmath.Geometry{}), nil
+	}
+
+	allGeometries := make([]spatialmath.Geometry, 0, len(ws.obstacles))
+	for _, gf := range ws.obstacles {
+		tf, err := fs.Transform(inputs, gf, World)
+		if err != nil {
+			return nil, err
 		}
-		return NewGeometriesInFrame(World, allGeometries), nil
+		allGeometries = append(allGeometries, tf.(*GeometriesInFrame).Geometries()...)
 	}
+	return NewGeometriesInFrame(World, allGeometries), nil
+}
 
-	if ws == nil {
-		ws = &WorldState{}
+func (ws *WorldState) rectifyNames(geometries []spatialmath.Geometry) ([]spatialmath.Geometry, error) {
```

Comment:
Can you add a description of what this function does for the future?

Reply:
I actually don't like that this is a helper function since it doesn't really standalone.  It doesn't actually modify the obstacles it only does half of what is necessary.  Its only being called in one place at this point so I will roll it into the constructor